lib/3scale_toolbox/configuration.rb

```diff
@@ -0,0 +1,38 @@
+require 'yaml/store'
+
+module ThreeScaleToolbox
+  class Configuration
+    def initialize(config_file)
+      @store_data = nil
+      @store = YAML::Store.new(config_file)
+    end
+
+    def data(key)
+      store_data[key]
+    end
+
+    def update(key)
+      return if key.nil?
+      # invalidate cache
+      @store_data = nil
+      @store.transaction do
+        @store[key] = yield @store[key]
+      end
+    end
+
+    private
+
+    def store_data
+      @store_data ||= read
```

Comment:
This makes it not threadsafe. IMO it is better to just open transaction for every read and keep the store to do its' job.